src/main/java/com/three_tech_solutions/slot_app/controllers/implementations/UserControllerImpl.java

```diff
@@ -19,11 +19,14 @@ public class UserControllerImpl implements UserController {
 
     private final UserService userService;
 
+
     @Override
-    public Page<StudentResponse> getUserStudents(UUID userId, String filter, Integer page, Integer size) {
+    public Page<StudentResponse> getUserStudents(UUID userId, String filter,String orderBy, String orderDirection, Integer page, Integer size) {
         return userService.getUserStudents(
                 userId,
                 filter,
+                orderBy,
+                orderDirection,
                 PageRequest.of(page, size)
         );
```

Comment:
Una de las pocas lógicas que es posible tener en los controllers, es la obtención de los datos (que vienen por parámetro) para enviarselo al service. En este caso, la validación del orderBy y orderDirection, es lógica que debería hacerse en el controller porque además, muchas veces esta logica se repite entre diferentes entidades que tienen paginación u ordenamiento.

De hecho, en el servicio vos estás generando el `Pageable` a partir del `Pageable` que ya estas mandando al userService mediante el `PageRequest.of(page, size)`. Aca podrias hacer lo siguiente

1. Crear este metodo
```
    private PageRequest buildPageable(String orderBy, String orderDirection, Integer page, Integer size) {
        return PageRequest.of(page, size, Sort.by(
                Sort.Direction.fromString(orderDirection),
                orderBy
        ));
    }
```

2. Llamar al `userService.getUserStudents()` enviando estos parametros

```suggestion
        return userService.getUserStudents(
                userId,
                filter,
                buildPageable(orderBy, orderDirection, page, size)
        );
```